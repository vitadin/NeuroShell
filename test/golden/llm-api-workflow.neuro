%% Test complete LLM API workflow: load → activate → load
%% This test demonstrates the behavior in a clean test environment

%% Step 1: Load initial state (no API keys in clean test environment)
\echo ═══ Initial State: No API Keys ═══
\llm-api-load

%% Step 2: Try to activate a non-existent key (should fail)
\echo ═══ Attempting to Activate Non-existent Key ═══
\try \llm-api-activate[provider=openai, key=os.OPENAI_API_KEY]
\get[_error]

%% Step 3: Load state after failed activation attempt
\echo ═══ State After Failed Activation ═══
\llm-api-load

%% Step 4: Test provider-specific filtering
\echo ═══ Testing Provider Filtering ═══
\llm-api-load[provider=openai]
\llm-api-load[provider=anthropic]

%% Step 5: Test invalid provider in activate command
\echo ═══ Testing Invalid Provider ═══
\try \llm-api-activate[provider=invalid, key=os.TEST_KEY]
\get[_error]