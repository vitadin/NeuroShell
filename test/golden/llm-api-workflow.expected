═══ Initial State: No API Keys ═══
No API keys found in any source.
═══ Attempting to Activate Non-existent Key ═══
_error = key 'os.OPENAI_API_KEY' is empty. Run \llm-api-show to see available keys
═══ State After Failed Activation ═══
No API keys found in any source.
═══ Testing Provider Filtering ═══
No API keys found for provider 'openai'.
No API keys found for provider 'anthropic'.
═══ Testing Invalid Provider ═══
_error = invalid provider 'invalid'. Valid providers: openai, anthropic, openrouter, moonshot, gemini